src/posix/platform/system.cpp

```diff
@@ -196,7 +196,10 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
     gInstance = otInstanceInitSingle();
     OT_ASSERT(gInstance != nullptr);
 
-    platformSetUp();
+    if (!gDryRun)
+    {
+        platformSetUp();
```

Comment:
Why do we need to disable `platformSetUp()` when the dry run is enabled? Do we also need to disable `platformTearDown()` when the dry run is enabled?

Reply:
Some uninitialized data might be used when setting up these modules. E.g. the interface name when creating the daemon `.lock` file.

I changed the PR to disable `platformTearDown()` as well.